File: backend/pokerogue/src/main/java/com/pokerogue/helper/ability/dto/AbilityDetailResponse.java
package com.pokerogue.helper.ability.dto;

import com.pokerogue.helper.ability.data.Ability;
import java.util.List;

public record AbilityDetailResponse(
        String koName,
        String description,
        List<AbilityPokemonResponse> pokemons
) {

    public static AbilityDetailResponse of(Ability ability, List<AbilityPokemonResponse> pokemons) {
        return new AbilityDetailResponse(ability.getName(), ability.getDescription(), pokemons);
    }
}

File: android/app/src/main/java/poke/rogue/helper/remote/dto/response/ability/AbilityDetailResponse.kt
package poke.rogue.helper.remote.dto.response.ability

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import poke.rogue.helper.remote.dto.response.pokemon.PokemonResponse

@Serializable
data class AbilityDetailResponse(
    val id: Long = 0,
    @SerialName("koName")
    val name: String,
    val description: String,
    val pokemons: List<PokemonResponse>,
)
